core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -254,7 +250,7 @@ public void testFromXContent() throws IOException {
         QB testQuery = createTestQueryBuilder();
         assertParsedQuery(testQuery.toString(), testQuery);
         for (Map.Entry<String, QB> alternateVersion : getAlternateVersions().entrySet()) {
-            assertParsedQuery(alternateVersion.getKey(), alternateVersion.getValue());
+            assertParsedQuery(alternateVersion.getKey(), alternateVersion.getValue(), ParseFieldMatcher.EMPTY);
```

Comment:
I think this should be strict too. The problem is that the not query has some alternate version that uses deprecated syntax. I think that should be moved to a separate test like you did with recent PRs.